src/shared/resolve-util.h

```diff
@@ -70,9 +71,21 @@ enum DnsOverTlsMode {
         _DNS_OVER_TLS_MODE_INVALID = -EINVAL,
 };
 
+enum DnsOverHttpsMode {
+        /* No connection is made for DNS-over-HTTPS */
+        DNS_OVER_HTTPS_NO,
+
+        /* Enforce DNS-over-HTTPS */
+        DNS_OVER_HTTPS_YES,
+
+        _DNS_OVER_HTTPS_MODE_MAX,
+        _DNS_OVER_HTTPS_MODE_INVALID = -EINVAL,
+};
```

Comment:
hmm, you modelled this, as if dot and doh were kinda orthogonal. is that really the best way to model this?

i.e. shouldn't we fold this into DnsOverTlsMode? i.e. have two (or three) more enums values there, maybe:

```c
enum DnsOverTlsMode {
        /* No connection is made for DNS-over-TLS nor DNS-over-HTTP */
        DNS_OVER_TLS_NO,
        /* Try to connect using DNS-over-TLS, but if connection fails,
         * fall back to using an unencrypted connection */
        DNS_OVER_TLS_OPPORTUNISTIC_TLS_ONLY,
        /* Enforce DNS-over-TLS and require valid server certificates */
        DNS_OVER_TLS_TLS_ONLY,
        /* Enforce DoH with good cert */ 
        DNS_OVER_TLS_HTTP_ONLY,
        /* Use either DoT or DoH with good certs, whatever works of the two */
        DNS_OVER_TLS_YES,
        /* Use DoH if it works, and otherwise classic DNS */
        DNS_OVER_TLS_OPPORTUNISTIC_HTTP_ONLY,
        /* Use DoH or DoT or classic DNS, whatever works */
        DNS_OVER_TLS_OPPORTUNISTIC,
        _DNS_OVER_TLS_MODE_MAX,
        _DNS_OVER_TLS_MODE_INVALID = -EINVAL,
};
```

Just wondering...

I mean DoH in a way is a superset of DoT, given that HTTPS is also using TLS...